components/shared/layouts/LoggedOutOnlyRoute/LoggedOutOnlyRoute.tsx

```diff
@@ -0,0 +1,28 @@
+import { useAuthContext } from "@context/Authentication";
+import { useRouter } from "next/router";
+import React, { ReactNode, useEffect } from "react";
+
+interface LoggedOutOnlyRouteProps {
+  children: ReactNode;
+}
+
+const LoggedOutOnlyRoute = ({ children }: LoggedOutOnlyRouteProps) => {
+  const { user, isLoading } = useAuthContext();
+  const router = useRouter();
+
+
+  useEffect(() => {
+    if (!isLoading && user) {
+        router.push("/");
+    }
+  }, [router, isLoading, user]);
+
+
+  if (!isLoading && !user) {
+    return <>{children}</>;
+  }
+
+  return <h1>Loading</h1>;
+};
+
+export default LoggedOutOnlyRoute;
```

Comment:
Hello Alex, just asking, have you tried this component in a real environment i.e wrapping the Login component with LoggedOutOnlyRoute? I'm just curious if the router.push from **logging in** will **conflict with the router.push** from **LoggedOutOnlyRoute** let me know :D